src/processTargets/modifiers/ItemStage/tokenizeRange.ts

```diff
@@ -0,0 +1,129 @@
+import { Range, TextEditor } from "vscode";
+
+export function tokenizeRange(
+  editor: TextEditor,
+  collectionRange: Range,
+  collectionBoundary?: [Range, Range]
+) {
+  const { document } = editor;
+  const text = document.getText(collectionRange);
+  const lexemes = text.split(/([,(){}<>[\]"'`]|\\"|\\'|\\`)/g).filter(Boolean);
+  const joinedLexemes = joinLexemesBySkippingMatchingPairs(lexemes);
+  const tokens: Token[] = [];
+  let offset = document.offsetAt(collectionRange.start);
+
+  joinedLexemes.forEach((lexeme) => {
+    // Whitespace found. Just skip
+    if (lexeme.trim().length === 0) {
+      offset += lexeme.length;
+      return;
+    }
+
+    // Separator delimiter found.
+    if (lexeme === delimiter) {
+      tokens.push({
+        type: "delimiter",
```

Comment:
Let's call this `"separator"`.  The term "delimiter" is too vague to be useful here, as you could easily interpret it to include boundaries